create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java

```diff
@@ -77,20 +79,30 @@ void shouldExtractScopusIdentifierAndPlaceItInsideAdditionalIdentifiersObject()
         CreatePublicationRequest createPublicationRequest = scopusHandler.handleRequest(s3Event, CONTEXT);
         var actualAdditionalIdentifiers = createPublicationRequest.getAdditionalIdentifiers();
         var expectedIdentifier =
-                new AdditionalIdentifier(ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME, SCP_ID_IN_0000469852);
+            new AdditionalIdentifier(ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME, SCP_ID_IN_0000469852);
         assertThat(actualAdditionalIdentifiers, contains(expectedIdentifier));
     }
 
+    @Test
+    void shouldExtractDoiAndPlaceItInsideReferenceObject() throws IOException {
+        var scopusFile = IoUtils.stringFromResources(Path.of(SCOPUS_XML_0000469852));
+        var uri = s3Driver.insertFile(UnixPath.of(randomString()), scopusFile);
+        S3Event s3Event = createS3Event(uri);
+        CreatePublicationRequest createPublicationRequest = scopusHandler.handleRequest(s3Event, CONTEXT);
+        URI expectedURI = new UriWrapper(DOI_OPEN_URL_FORMAT).addChild(DOI_IN_0000469852).getUri();
+        assertThat(createPublicationRequest.getEntityDescription().getReference().getDoi(), equalToObject(expectedURI));
+    }
```

Comment:
That depends on correct implementation of equals in URI class. Jeg ville har sjekket uri.toString() mot string. Men er det bedre?

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URI.html#equals(java.lang.Object)
Seems like the equals method works in the way it is need for this test to work properly